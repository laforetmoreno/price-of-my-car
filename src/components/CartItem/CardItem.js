import React from 'react';
import { string, bool } from 'prop-types';

import style from './CardItem.module.scss';

const CardItem = ({ label, field, price }) => (
  <li className={style.item}>
    <span>{label}</span>
    <strong>{price ? `R$ ${field}` : field}</strong>
  </li>
);

CardItem.propTypes = {
  label: string,
  field: string,
  price: bool,
};

export default CardItem;
