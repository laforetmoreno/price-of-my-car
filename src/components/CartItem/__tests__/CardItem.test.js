import React from 'react';
import renderer from 'react-test-renderer';
import CardItem from '../CardItem';

describe('<CardItem />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CardItem label="test" field="test" price />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
