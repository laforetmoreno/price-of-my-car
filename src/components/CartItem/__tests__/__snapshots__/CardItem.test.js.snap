// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> renders correctly 1`] = `
<li
  className="item"
>
  <span>
    test
  </span>
  <strong>
    R$Â NaN
  </strong>
</li>
`;
